tests/bluetooth/bsim/mesh/src/test_dfu.c

```diff
@@ -764,12 +859,125 @@ static void test_cli_fail_on_persistency(void)
 	PASS();
 }
 
+static void test_cli_stop(void)
+{
+	int err;
+
+	(void)target_srv_add(TARGET_ADDR + 1, true);
+
+	switch (expected_stop_phase) {
+	case BT_MESH_DFU_PHASE_TRANSFER_ACTIVE:
+		cli_common_fail_on_init();
+
+		dfu_cli_xfer.xfer.mode = BT_MESH_BLOB_XFER_MODE_PUSH;
+
+		err = bt_mesh_dfu_cli_send(&dfu_cli, &dfu_cli_xfer.inputs, &dummy_blob_io,
+					   &dfu_cli_xfer.xfer);
+		if (err) {
+			FAIL("DFU Client send failed (err: %d)", err);
+		}
+
+		if (k_sem_take(&dfu_started, K_SECONDS(200))) {
+			FAIL("Firmware transfer failed");
+		}
+
+		ASSERT_EQUAL(BT_MESH_DFU_ERR_INTERNAL, dfu_cli_xfer.targets[0].status);
+		ASSERT_EQUAL(BT_MESH_DFU_PHASE_TRANSFER_ACTIVE, dfu_cli_xfer.targets[0].phase);
+		break;
+	case BT_MESH_DFU_PHASE_VERIFY:
+		cli_common_init_recover();
+
+		dfu_cli_xfer.xfer.mode = BT_MESH_BLOB_XFER_MODE_PUSH;
+
+		err = bt_mesh_dfu_cli_send(&dfu_cli, &dfu_cli_xfer.inputs, &dummy_blob_io,
+					   &dfu_cli_xfer.xfer);
+		if (err) {
+			FAIL("DFU Client resume failed (err: %d)", err);
+		}
+
+		if (k_sem_take(&dfu_verifying, K_SECONDS(200))) {
+			FAIL("Firmware transfer failed");
+		}
+		ASSERT_EQUAL(BT_MESH_DFU_ERR_INTERNAL, dfu_cli_xfer.targets[0].status);
+		ASSERT_EQUAL(BT_MESH_DFU_PHASE_VERIFY, dfu_cli_xfer.targets[0].phase);
+
+		break;
+	case BT_MESH_DFU_PHASE_VERIFY_OK:
+		/* Nothing to do here on distributor side, target must verify image */
+		break;
+	case BT_MESH_DFU_PHASE_VERIFY_FAIL:
+		cli_common_fail_on_init();
+
+		dfu_cli_xfer.xfer.mode = BT_MESH_BLOB_XFER_MODE_PUSH;
+
+		err = bt_mesh_dfu_cli_send(&dfu_cli, &dfu_cli_xfer.inputs, &dummy_blob_io,
+					   &dfu_cli_xfer.xfer);
+		if (err) {
+			FAIL("DFU Client send failed (err: %d)", err);
+		}
+
+		if (k_sem_take(&dfu_verify_failed, K_SECONDS(200))) {
+			FAIL("Firmware transfer failed");
+		}
+
+		ASSERT_EQUAL(BT_MESH_DFU_ERR_WRONG_PHASE, dfu_cli_xfer.targets[0].status);
+		ASSERT_EQUAL(BT_MESH_DFU_PHASE_VERIFY_FAIL, dfu_cli_xfer.targets[0].phase);
+		break;
+	case BT_MESH_DFU_PHASE_APPLYING:
+		cli_common_init_recover();
+
+		dfu_cli_xfer.xfer.mode = BT_MESH_BLOB_XFER_MODE_PUSH;
+
+		err = bt_mesh_dfu_cli_send(&dfu_cli, &dfu_cli_xfer.inputs, &dummy_blob_io,
+					   &dfu_cli_xfer.xfer);
+
+		if (err) {
+			FAIL("DFU Client send failed (err: %d)", err);
+		}
+		if (k_sem_take(&dfu_ended, K_SECONDS(200))) {
+			FAIL("Firmware transfer failed");
+		}
+
+		bt_mesh_dfu_cli_apply(&dfu_cli);
+		if (k_sem_take(&dfu_cli_applied_sem, K_SECONDS(200))) {
+			/* This will time out as target will reboot before applying */
+		}
+		ASSERT_EQUAL(BT_MESH_DFU_ERR_INTERNAL, dfu_cli_xfer.targets[0].status);
+		ASSERT_EQUAL(BT_MESH_DFU_PHASE_APPLYING, dfu_cli_xfer.targets[0].phase);
+		break;
+	case BT_MESH_DFU_PHASE_APPLY_SUCCESS:
+		cli_common_init_recover();
+
+		dfu_cli.xfer.state = 5;
+		dfu_cli.xfer.slot = dfu_cli_xfer.xfer.slot;
+		dfu_cli.xfer.blob.id = blob_id;
```

Comment:
Is `blob_id` initialized anywhere other than at start with zero value? Or should it be `TEST_BLOB_ID`?

Reply:
It should be `TEST_BLOB_ID`